server/clip_server/executors/helper.py

```diff
@@ -24,20 +25,27 @@ def preproc_image(
 ) -> Tuple['DocumentArray', Dict]:
 
     tensors_batch = []
+    filtered_da = DocumentArray()
 
     for d in da:
         content = d.content
 
-        if d.blob:
-            d.convert_blob_to_image_tensor()
-        elif d.tensor is None and d.uri:
-            # in case user uses HTTP protocol and send data via curl not using .blob (base64), but in .uri
-            d.load_uri_to_image_tensor()
+        try:
+            if d.blob:
+                d.convert_blob_to_image_tensor()
+            elif d.tensor is None and d.uri:
+                # in case user uses HTTP protocol and send data via curl not using .blob (base64), but in .uri
+                d.load_uri_to_image_tensor()
 
-        tensors_batch.append(preprocess_fn(d.tensor).detach())
+        except Exception:
+            warnings.warn(f'Failed to load image doc with ID {d.id}.')
+            continue
 
-        # recover doc content
-        d.content = content
+        else:
+            tensors_batch.append(preprocess_fn(d.tensor).detach())
+            filtered_da.append(d)
+            # recover doc content
+            d.content = content
```

Comment:
just move this block into the `try ... except`

Reply:
ok, `tensors_batch.append(preprocess_fn(d.tensor).detach())
            filtered_da.append(d)` will always work right?

Reply:
ok, `tensors_batch.append(preprocess_fn(d.tensor).detach())
            filtered_da.append(d)` will always work right?